src/lib/steps.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * This module defines the supported unit-of-transformation steps.
+ */
+
+export interface DomainStep {
```

Comment:
Rename to `selectDataset` I guess following previous comments

Reply:
:+1: I will do this in a distinct PR once this one is merged.